newrelic/hooks/datastore_motor.py

```diff
@@ -12,35 +12,131 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from newrelic.api.datastore_trace import DatastoreTrace
+from newrelic.api.function_trace import wrap_function_trace
 from newrelic.common.object_wrapper import wrap_function_wrapper
 
-# This is NOT a fully-featured instrumentation for the motor library. Instead
-# this is a monkey-patch of the motor library to work around a bug that causes
-# the __name__ lookup on a MotorCollection object to fail. This bug was causing
-# customer's applications to fail when they used motor in Tornado applications.
+_motor_client_sync_methods = (
+    "aggregate_raw_batches",
+    "aggregate",
+    "find_raw_batches",
+    "find",
+    "list_indexes",
+    "list_search_indexes",
+    "watch",
+)
 
+_motor_client_async_methods = (
+    "bulk_write",
+    "count_documents",
+    "create_index",
+    "create_indexes",
+    "create_search_index",
+    "create_search_indexes",
+    "delete_many",
+    "delete_one",
+    "distinct",
+    "drop_index",
+    "drop_indexes",
+    "drop_search_index",
+    "drop",
+    "estimated_document_count",
+    "find_one_and_delete",
+    "find_one_and_replace",
+    "find_one_and_update",
+    "find_one",
+    "index_information",
+    "insert_many",
+    "insert_one",
+    "options",
+    "rename",
+    "replace_one",
+    "update_many",
+    "update_one",
+    "update_search_index",
+)
 
-def _nr_wrapper_Motor_getattr_(wrapped, instance, args, kwargs):
 
-    def _bind_params(name, *args, **kwargs):
-        return name
+def instance_info(collection):
+    nodes = collection.database.client.nodes
```

Comment:
Should we use getattr/hasattr here in case these attr's don't exist? There's a couple place where we should probably do this but I'm just noting one occurrence here.